pkg/sbom/spdx/marshal.go

```diff
@@ -282,7 +282,7 @@ func (m *Marshaler) pkgToSpdxPackage(t, pkgDownloadLocation string, class types.
 	}
 
 	var pkgSrcInfo string
-	if class == types.ClassOSPkg {
+	if class == types.ClassOSPkg && pkg.SrcName != "" {
```

Comment:
Would you elaborate on that?

Reply:
Some package does not have srcName.
e.g. 

```
        {
          "ID": "gpg-pubkey@c87f5b1a-593863f8.",
          "Name": "gpg-pubkey",
          "Version": "c87f5b1a",
          "Release": "593863f8",
          "Arch": "None",
          "Licenses": [
            "pubkey"
          ],
          "Layer": {}
        },
```

In this case, spdx output empty SourceInfo.

```
##### Package: gpg-pubkey

PackageName: gpg-pubkey
SPDXID: SPDXRef-Package-6c5513710c857f3f
PackageVersion: c87f5b1a-593863f8
PackageDownloadLocation: NONE
FilesAnalyzed: false
PackageSourceInfo: built package from: 
PackageLicenseConcluded: pubkey
PackageLicenseDeclared: pubkey
ExternalRef: PACKAGE-MANAGER purl pkg:rpm/amazon/gpg-pubkey@c87f5b1a-593863f8?arch=None&distro=amazon-2%20%28Karoo%29
PackageAttributionText: PkgID: gpg-pubkey@c87f5b1a-593863f8.
```